app/pages/admin.js

```diff
@@ -46,16 +70,39 @@ export default function Admin() {
                 </div>
                 <div className="flex w-screen mx-4 mt-2">
                   <div>
-                    {links.map((link, index) => (
-                      <div key={index}>
-                        {amountOfLinks > 0 &&  (
-                        <div>
-                          <span>Name</span>
-                          <input name="name" type="text" id="name" required  onBlur={change(index)}/>
-                          <span>Link</span>
-                          <input name="link" type="text" id="link" required />
-                        </div>
-                          )}
+                    {[...links.keys()].map((key) => (
```

Comment:
Couple of comments here:
1. No need to spread array of .keys() since that is already a newly created array
2. .entries is more suitable since you need both key and value from the links Map, if you were to change from Map to object this would change to `Object.entries(links).map(...)`
```suggestion
                    {links.entries().map(([key, value]) => (
```

Reply:
Change done, but adding with the add button needs to be fixed.